src/runtime/storageNG/tests/firebase-store-integration-test.ts

```diff
@@ -0,0 +1,192 @@
+    assert.deepEqual(firebaseEntry.model, activeStore1['localModel'].getData());
+    assert.equal(firebaseEntry.version, 3);
+  });
+
+  it('will store operation updates from multiple sources with some delays', async () => {
+    const runtime = new Runtime();
+    const storageKey = new MockFirebaseStorageKey('unique');
+    const store1 = new Store<CRDTCountTypeRecord>(storageKey, Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+    const activeStore1 = await store1.activate();
+
+    const store2 = new Store<CRDTCountTypeRecord>(storageKey, Exists.ShouldExist, null, StorageMode.Direct, CRDTCount);
+    const activeStore2 = await store2.activate();
+
+    void activeStore1.onProxyMessage({type: ProxyMessageType.Operations, id: 1, operations: [
+      {type: CountOpTypes.Increment, actor: 'me', version: {from: 0, to: 1}}
+    ]});
+    void activeStore1.onProxyMessage({type: ProxyMessageType.Operations, id: 1, operations: [
+      {type: CountOpTypes.Increment, actor: 'them', version: {from: 0, to: 1}}
+    ]});
+
+    await 0;
+    await 0;
+
+    void activeStore2.onProxyMessage({type: ProxyMessageType.Operations, id: 1, operations: [
+      {type: CountOpTypes.Increment, actor: 'other', version: {from: 0, to: 1}}
+    ]});
+
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+
+    void activeStore2.onProxyMessage({type: ProxyMessageType.Operations, id: 1, operations: [
+      {type: CountOpTypes.Increment, actor: 'other', version: {from: 1, to: 2}}
+    ]});
+
+    await activeStore1.idle();
+    await activeStore2.idle();
+
+    assert.deepEqual(activeStore1['localModel'], activeStore2['localModel']);
+  });
+
+  it(`doesn't deadlock given a particular timing pattern`, async () => {
+    const runtime = new Runtime();
+    const storageKey = new MockFirebaseStorageKey('unique');
+    const store1 = new Store<CRDTCountTypeRecord>(storageKey, Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+    const activeStore1 = await store1.activate();
+
+    const store2 = new Store<CRDTCountTypeRecord>(storageKey, Exists.ShouldExist, null, StorageMode.Direct, CRDTCount);
+    const activeStore2 = await store2.activate();
+
+    void activeStore1.onProxyMessage({type: ProxyMessageType.Operations, id: 1, operations: [
+      {type: CountOpTypes.Increment, actor: 'me', version: {from: 0, to: 1}}
+    ]});
+    void activeStore1.onProxyMessage({type: ProxyMessageType.Operations, id: 1, operations: [
+      {type: CountOpTypes.Increment, actor: 'them', version: {from: 0, to: 1}}
+    ]});
+    void activeStore2.onProxyMessage({type: ProxyMessageType.Operations, id: 1, operations: [
+      {type: CountOpTypes.Increment, actor: 'other', version: {from: 0, to: 1}}
+    ]});
+
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
+    await 0;
```

Comment:
Given the 'particular timing pattern' in the test description, I presume the number of awaits is specific in this test? Might be worth a comment explaining how the numbers are derived.